### PR TITLE
[Swift] various dead code cleanups, simplifications and phpdoc fixes.

### PR DESCRIPTION
**Pull request by hhamon**

### CODE REVIEW

File: lib/classes/Swift/CharacterStream/NgCharacterStream.php [hunk 1]
```diff
@@ -178,10 +178,6 @@ public function read($length)
                 break;
 
             case Swift_CharacterReader::MAP_TYPE_INVALID:
-                $end = $this->_currentPos + $length;
```
Comment (hhamon) [inline lib/classes/Swift/CharacterStream/NgCharacterStream.php]: Note for reviewers: unused variable `$end`

File: lib/classes/Swift/CharacterStream/NgCharacterStream.php [hunk 2]
```diff
@@ -178,10 +178,6 @@ public function read($length)
...
-                $end = $end > $this->_charCount
```
Comment (hhamon) [inline lib/classes/Swift/CharacterStream/NgCharacterStream.php]: Note for reviewers: unused variable `$end`

File: lib/classes/Swift/Signers/DKIMSigner.php [hunk 3]
```diff
@@ -206,7 +199,6 @@ public function reset()
     {
         $this->_headerHash = null;
         $this->_signedHeaders = array();
-        $this->_headerHashHandler = null;
```
Comment (hhamon) [inline lib/classes/Swift/Signers/DKIMSigner.php]: Note for reviewers: unused property `$_headerHashHandler`

File: lib/classes/Swift/Signers/DKIMSigner.php [hunk 4]
```diff
@@ -143,13 +143,6 @@ class Swift_Signers_DKIMSigner implements Swift_Signers_HeaderSigner
      */
     protected $_dkimHeader;
 
-    /**
-     * Hash Handler.
-     *
-     * @var hash_ressource
-     */
-    private $_headerHashHandler;
```
Comment (hhamon) [inline lib/classes/Swift/Signers/DKIMSigner.php]: Note for reviewers: unused property `$_headerHashHandler`

File: lib/classes/Swift/Signers/SMimeSigner.php [hunk 5]
```diff
@@ -400,7 +400,6 @@ protected function streamToMime(Swift_OutputByteStream $fromStream, Swift_Messag
             }
 
             $boundary = trim($contentTypeData['1'], '"');
-            $boundaryLen = strlen($boundary);
```
Comment (hhamon) [inline lib/classes/Swift/Signers/SMimeSigner.php]: Note for reviewers: unused property `$boundaryLen`

File: lib/classes/Swift/Transport/AbstractSmtpTransport.php [hunk 6]
```diff
@@ -417,7 +417,7 @@ private function _doMailTransaction($message, $reversePath, array $recipients, a
         foreach ($recipients as $forwardPath) {
             try {
                 $this->_doRcptToCommand($forwardPath);
-                $sent++;
+                ++$sent;
```
Comment (hhamon) [inline lib/classes/Swift/Transport/AbstractSmtpTransport.php]: Note for reviewers: as requested by fabbot.io.

File: lib/classes/Swift/Transport/FailoverTransport.php [hunk 7]
```diff
@@ -23,14 +23,6 @@ class Swift_Transport_FailoverTransport extends Swift_Transport_LoadBalancedTran
     private $_currentTransport;
 
     /**
-     * Creates a new FailoverTransport.
-     */
-    public function __construct()
```
Comment (hhamon) [inline lib/classes/Swift/Transport/FailoverTransport.php]: Note for reviewers: no need for an explicit empty constructor.

File: lib/classes/Swift/Transport/LoadBalancedTransport.php [hunk 8]
```diff
@@ -30,13 +30,6 @@ class Swift_Transport_LoadBalancedTransport implements Swift_Transport
     protected $_transports = array();
 
     /**
-     * Creates a new LoadBalancedTransport.
-     */
-    public function __construct()
```
Comment (hhamon) [inline lib/classes/Swift/Transport/LoadBalancedTransport.php]: Note for reviewers: no need for an explicit empty constructor.

File: lib/classes/Swift/Transport/SendmailTransport.php [hunk 9]
```diff
@@ -102,6 +102,7 @@ public function send(Swift_Mime_Message $message, &$failedRecipients = null)
         $failedRecipients = (array) $failedRecipients;
         $command = $this->getCommand();
         $buffer = $this->getBuffer();
+        $count = 0;
```
Comment (hhamon) [inline lib/classes/Swift/Transport/SendmailTransport.php]: Note for reviewers: variable may not have been defined and initialized later in the code.

**Comment by hhamon** [comment]
I don't know why the tests suite ends up with a segfault on PHP 5.3. On my machine (PHP 5.6.12), the test suite also segfaults when I run unit tests after a fresh checkout of the repository + composer install command.

**Comment by sstok** [comment]
PHP 5.3 is pretty dead on TravisCi, and try disabling the gc as used in https://github.com/symfony/phpunit-bridge/blob/2.7/bootstrap.php#L11-L15

File: lib/classes/Swift/Plugins/DecoratorPlugin.php [hunk 10]
```diff
@@ -157,12 +157,12 @@ public function getReplacementsFor($address)
     {
         if ($this->_replacements instanceof Swift_Plugins_Decorator_Replacements) {
             return $this->_replacements->getReplacementsFor($address);
-        } else {
-            return isset($this->_replacements[$address])
+        }
+
+        return isset($this->_replacements[$address])
                 ? $this->_replacements[$address]
                 : null
```
Comment (stof) [inline lib/classes/Swift/Plugins/DecoratorPlugin.php]: the indentation should be 4 spaces here, not 8

Comment (hhamon) [inline lib/classes/Swift/Plugins/DecoratorPlugin.php]: Certainly! There are many other locations in the code with that kind of weird indentations.

Comment (hhamon) [inline lib/classes/Swift/Plugins/DecoratorPlugin.php]: fixed.

**Comment by hhamon** [comment]
@stof also any reasons why the build matrix allows failures for PHP 5.6? Shouldn't we officially support PHP5.6 with Swiftmailer?

**Comment by stof** [comment]
@hhamon no idea

**Comment by hhamon** [comment]
So we can add PHP 5.6 to the list of supported versions I guess.

File: lib/classes/Swift/Signers/DKIMSigner.php [hunk 11]
```diff
@@ -588,9 +580,13 @@ protected function _addHeader($header, $is_sig = false)
         $this->_addToHeaderHash($header);
     }
 
+    /**
+     * This method is currently useless in this class but it must be kept for BC
+     * reasons due to its "protected" scope. This method might overriden by
+     * custom client code.
+     */
```
Comment (fabpot) [inline lib/classes/Swift/Signers/DKIMSigner.php]: You should add a `@deprecated` tag instead.

**Comment by fabpot** [comment]
Probably because 5.6 did not work back then. It should indeed be re-integrated into the main matrix, and hopefully PHP7 as well.

**Comment by fabpot** [comment]
Thank you @hhamon.

**Pull request merged**